sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala

```diff
@@ -248,4 +249,29 @@ class AnalysisSuite extends AnalysisTest {
     val plan = relation.select(CaseWhen(Seq((Literal(true), 'a.attr)), 'b).as("val"))
     assertAnalysisSuccess(plan)
   }
+
+  test("check resolve of natural join") {
```

Comment:
maybe you should separate this into its own suite, and create different test cases.

it is getting pretty hard to understand what this is testing.